docs/source/developer/overview.rst

```diff
@@ -0,0 +1,112 @@
+Overview
+========
+
+This page aims to introduce the main classes and modules of the ``metasyn`` package. It is intended to be a high-level overview of the package's structure and functionality. For a thorough overview, please refer to the :doc:`API reference </api/metasyn>`. Clicking on a class or module name will automatically take you to its API reference section.
+
+.. warning:: 
+    This page is intended for developers and advanced users. If you are new to ``metasyn``, please refer to the :doc:`/usage/usage` first. 
+
+
+MetaFrame class
+---------------
+
+The :class:`~metasyn.MetaFrame` class is a core component of the ``metasyn`` package. It represents a metadata frame, which is a structure that holds statistical metadata about a dataset. The data contained in a :obj:`~metasyn.MetaFrame` is in line with the :doc:`/developer/GMF`.
+
+**Attributes:**
+
+- ``meta_vars``: A list of :obj:`~metasyn.MetaVar` objects, each representing a column in a DataFrame.
+- ``n_rows``: The number of rows in the original DataFrame.
+
+**Properties:**
+
+- :obj:`~metasyn.MetaFrame.n_columns`: Returns the number of columns in the original DataFrame.
+- :obj:`~metasyn.MetaFrame.descriptions`: Returns a dictionary of column descriptions. Can also be set to update the descriptions.
+
+**Methods:**
```

Comment:
I'm starting to worry a little bit that this is too much of a copy of the automatically generated API documentation. Maybe it's better to just highlight a subset of the methods, etc, and leave the exhaustive list to the automatically generated documentation?

Reply:
Yeah I agree, I feel the same. I do think it's nice to expand on the docstrings though, to make them more easy to understand and more elaborate - that way the automatically generated API will be more complete